### ISSUE TITLE
jax.grad returns inf for sqrt'(smallest positive float64) where finite value is expected

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#39799]
### Description

`jax.grad` appears to return `inf` for the gradient of `jax.numpy.sqrt` at the smallest positive `float64` value.

The derivative of `sqrt(x)` is:

```
sqrt'(x) = 1 / (2 * sqrt(x))
```

For `x = 5e-324`, the smallest positive subnormal `float64`, the true derivative is approximately `2.2494568972715982e+161`, which is finite and representable in `float64`. However, JAX returns `inf`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

x = 5e-324
expected = 2.2494568972715982e+161

out = float(jax.grad(lambda z: jnp.sqrt(z))(jnp.asarray(x, dtype=jnp.float64)))

print("jax:", out)
print("expected:", expected)

if not (math.isfinite(out) and abs(out - expected) <= 1e-6 * abs(expected)):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: inf
expected: 2.2494568972715982e+161
```

### Expected result

```
2.2494568972715982e+161
```

The expected derivative is large but finite and within the representable `float64` range. JAX instead returns `inf`, suggesting that the backward rule overflows internally for the smallest positive subnormal input.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
#38651
